apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,246 @@
+        case 0:
+          return `${Math.floor(interval.hours)} hours`;
+        case 1:
+          return `${interval.days} day`;
+        default:
+          return `${interval.days} days`;
+      }
+    }
+    return (
+      (interval.hours ? interval.hours + 'h' : '') +
+      (interval.minutes ? ' ' + interval.minutes + 'm' : '')
+    );
+  };
+
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
+    const localArrival = route && DateTime.fromISO(route.arrival?.localTime);
+    const localDeparture =
+      route && DateTime.fromISO(route.departure?.localTime);
+    const utcArrival = route && DateTime.fromISO(route.arrival?.utcTime);
+    const utcDeparture = route && DateTime.fromISO(route.departure?.utcTime);
+    const duration =
+      route &&
+      this.intervalToString(this.getDateTimeInterval(utcArrival, utcDeparture));
+    return {
+      arrival: (route && route.arrival?.city) ?? '',
+      arrivalTime:
+        (localArrival && localArrival.toLocaleString(DateTime.TIME_SIMPLE)) ??
+        '',
+      carrier: { code: (route && route.airline) ?? '', name: '' }, // @TODO: get name of the airline
+      tripDate: (localDeparture && localDeparture.toFormat(dateFormat)) ?? '',
+      departure: (route && route.departure?.city) ?? '',
+      departureTime:
+        (localDeparture &&
+          localDeparture.toLocaleString(DateTime.TIME_SIMPLE)) ??
+        '',
+      duration: duration ?? '',
+      id: (route && route.id) ?? '',
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
+    if (!route) {
+      return null;
+    }
+    return route.findIndex(
+      routeItem => routeItem?.departure?.cityCode === routes?.[1]?.[0],
+    );
+  };
+
+  // @TODO: make ConnectionCard more universal (not only for forth and back routes)
+  getSanitizedRoute = (): Array<TripSectorWithId> => {
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    route: Array<TripSectorWithId>, // rename to sanitizeRoute
+    demandedSector: number,
+    sectorBorderIndex: ?number,
+  ): Array<TripSectorWithId> => {
+    if (
+      demandedSector === null ||
+      sectorBorderIndex === null ||
+      sectorBorderIndex === -1
+    ) {
+      return route;
+    }
+    if (demandedSector === 0) {
+      return route.slice(0, sectorBorderIndex);
+    }
+    return route.slice(sectorBorderIndex);
+  };
+
+  getDurationBetweenSectors = (route: ?Route, sectorBorderIndex: ?number) => {
+    if (route && sectorBorderIndex != null && sectorBorderIndex > 0) {
+      const stopBeforeSector = route[sectorBorderIndex - 1];
+      const stopAfterSector = route[sectorBorderIndex];
+
+      const utcDeparture =
+        stopAfterSector &&
+        stopAfterSector.departure &&
```

Comment:
I guess you could shorten it to `stopAfterSector.?departure && ...`

Reply:
It doesn't work. I guess it handles case only when it's `undefined` but not when it's `null`. Or something similar.

Reply:
I suppose that it should work with `undefined` and `null`
https://github.com/tc39/proposal-optional-chaining

Reply:
I see, so there could be some other problem. 

<img width="780" alt="screenshot 2019-01-14 at 14 17 19" src="https://user-images.githubusercontent.com/858321/51114975-2ff60200-1807-11e9-8bc3-f5a10632e9be.png">